### ISSUE TITLE
bugwarrior-pull from github closes unrelated tasks

### ISSUE THREAD

**Issue by er4z0r** [ralphbean/bugwarrior#219]
HI,

I just started with bugwarrior using the current devel-branch. When I run bugwarrior-pull it successfully pulls my issues from github but then closes 8 tasks that are completely unrelated to github. All theses tasks have in common is that they have URL annotations.  Here is my bugwarriorrc

```
[general]
targets = my_github 
inline_links = False
annotation_links = True
legacy_matching=False
log.level = DEBUG
[hooks]

[my_github]
service = github
default_priority = H
add_tags = open_source
github.username = er4z0r
github.token = <mytoken>
github.login = er4z0r
password = @oracle:use_keyring

```

And here my task diagnostics output:

```
task 2.4.3
   Platform: Darwin

Compiler
    Version: 4.2.1 Compatible Apple LLVM 6.1.0 (clang-602.0.49)
       Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
 Compliance: C++11

Build Features
      Built: Apr 21 2015 15:53:46
     Commit: 9959d8e
      CMake: 3.2.2
       Caps: -pthreads +tls
    libuuid: libuuid + uuid_unparse_lower
  libgnutls: 3.3.14
 Build type: Release

Configuration
       File: /Users/xxx/.taskrc (found), 2081 bytes, mode 100600
       Data: /Users/xxx/.task (found), dir, mode 40700
    Locking: Enabled
         GC: Enabled
    $EDITOR: mvim
     Server: xxx.xxx.xxx.xxx:xxx
         CA: ... (readable)
      Trust: strict
Certificate: ... (readable)
        Key: ... (readable)
    Ciphers: NORMAL
      Creds: Public/er4z0r/************************************

Hooks
    Scripts: Enabled
             (-none-)

Tests
      $TERM: xterm-256color (314x64)
       Dups: Scanned 91 tasks for duplicate UUIDs:
             No duplicates found
```

how can I prevent this?

**Comment by ralphbean** [comment]
Can you share some information about one or more of the tasks?  It might help to see more of what\'s in the annotations, etc.

I'll say that I'm very surprised that the legacy_matching parameter didn't make a difference.

**Comment by er4z0r** [comment]
Sure thing. Here are two of the affected tasks.

```
task 31 1                                                                                                                                                                           [16:05:23]
No command specified - assuming 'information'.

Name          Value
ID            1
Description   write habitrpg- bootstrap
                2015-03-05 14:17:20 https://github.com/elssar/pyhabit
                2015-03-05 14:17:21 http://tasklib.readthedocs.org/en/latest/
Status        Pending
Entered       2015-03-05 14:17:20 (10 weeks)
Last modified 2015-03-05 14:17:20 (10 weeks)
Tags          weekend fun coding
Virtual tags  ANNOTATED PENDING READY TAGGED UNBLOCKED
UUID          3197272a-6054-4aab-85d5-387165dc9269
Urgency       2.316

    annotations    0.9 *    1 =    0.9
    tags             1 *    1 =      1
    age          0.208 *    2 =  0.416
                                ------
                                 2.316

Date                Modification
2015-03-05 14:24:35 Tags changed from 'weekend' to 'weekend,fun'.
2015-03-05 14:24:55 Tags changed from 'weekend,fun' to 'weekend,fun,coding'.
2015-03-05 20:56:19 Start set to '2015-03-05 20:56:19'.
2015-03-05 22:13:31 Start deleted (duration: 1:17:12).
2015-03-06 09:19:18 Annotation of 'https://github.com/elssar/pyhabit' added.
2015-03-06 09:19:46 Annotation of 'http://tasklib.readthedocs.org/en/latest/' added.
2015-03-22 12:21:28 Start set to '2015-03-22 12:21:28'.
2015-03-22 19:30:10 Start deleted (duration: 7:08:42).
2015-04-01 17:16:41 Annotation 'https://github.com/elssar/pyhabit' deleted.
                    Annotation 'http://tasklib.readthedocs.org/en/latest/' deleted.
                    Annotation of 'https://github.com/elssar/pyhabit' added.
                    Annotation of 'http://tasklib.readthedocs.org/en/latest/' added.


Name          Value
ID            31
Description   Checkout Shadow Daemon
                2015-04-27 18:42:35 https://shadowd.zecure.org/overview/introduction/
Status        Pending
Entered       2015-04-27 18:42:24 (3 weeks)
Last modified 2015-04-27 18:42:35 (3 weeks)
Tags          mothership
Virtual tags  ANNOTATED PENDING READY TAGGED UNBLOCKED
UUID          1047ff98-4079-41d3-b348-0e9ca688af2e
Urgency       1.721

    annotations    0.8 *    1 =    0.8
    tags           0.8 *    1 =    0.8
    age           0.06 *    2 =  0.121
                                ------
                                 1.721

Date                Modification
2015-04-27 19:26:47 Annotation of 'https://shadowd.zecure.org/overview/introduction/' added.
```

**Comment by ralphbean** [comment]
I don't think this is still an issue based on all the changes since it was filed with respect to UDAs.

If I'm wrong and you're still hitting the issue, please re-open or file a new issue @er4z0r!
